### ISSUE TITLE
Refactor: CV pipeline

### ISSUE THREAD

**Issue by hi-liang** [arenaxr/arena-web-core#612]
Currently armarker system initializes the device-depenedent cv pipeline exclusively to the apriltag detector

What should happen is that the cv pipeline belongs to its own system which allows (potentially multiple) cv subscribers for frame data. This will allow each subscriber to pull frames as often as needed

Probably implement w/ event being emitted every frame (unknown effect on timing as these enter global async queue) with a getter function for last pose-synced frame.

**Comment by nampereira** [comment]
Yup, sounds good! At the time, I thought about it, but there was no real need for it.

The pipeline was sorta created with it in mind, but I admit there might be some issues to think through. The biggest one, I think, is that the current pipeline relies on a single worker to signal "I'm done, please get another frame" and only has a single buffer that is passed between camera capture and the worker. 

Otherwise, each camera capture mechanism (in camera-capture) has a `setCVWorker` that could be turned into an `addCVWorker` to a list of workers. Then, when a new frame is available, the postMessage could send the pixels to several workers (I think the underlying buffer allows sharing). The issue is really how to trigger the next frame capture when we have multiple workers.

**Comment by hi-liang** [comment]
That's a good point, and I think the array is exclusive to the main thread or a single worker at either stage. So different workers can't simultaneously share the same resource since its zero-copy transfer of the ArrayBuffer, they would be blocking to each other and the next camera frame.

So subscription could be more on a per-frame level.

1. WorkerA: getFrame msg, adds to buffer dispatch queue
2. Main - readPixels to Buffer
3. Main - Dispatch Buffer to: workerA
4. WorkerB: getFrame msg
5. WorkerA: getFrame msg (might actually get in anytime from before Main's readPixels process until Buffer comes back from WorkerB) 
6. Main - readPixels to Buffer
7. Dispatch Buffer to: workerB
8. Dispatch Buffer to: workerA

**Comment by nampereira** [comment]
Yeah think that should work, particularly if workers all use the shared frame for a similar amount of time. If all workers are Wasm, it makes sense to always do something like this: copy into its memory, let camera capture pass the frame along and only then process.

For example, we could split [processGsFrame()](https://github.com/arenaxr/arena-web-core/blob/a0428afb1507cc7fd4f843eed0e62cea5dbd0519/src/systems/armarker/apriltag-detector/apriltag.js#L265C16-L265C30) in the current worker to copy the buffer to wasm land and release it before processing (processing is done on the copied buffer when we call [`apritlTag.detect()`](https://github.com/arenaxr/arena-web-core/blob/a0428afb1507cc7fd4f843eed0e62cea5dbd0519/src/systems/armarker/apriltag-detector/apriltag.js#L273)).
